internal/authn/authinfo/db.go

```diff
@@ -0,0 +1,50 @@
+package authinfo
+
+import (
+	"context"
+	"errors"
+
+	"github.com/glasskube/distr/internal/apierrors"
+	"github.com/glasskube/distr/internal/authn"
+	"github.com/glasskube/distr/internal/db"
+	"github.com/glasskube/distr/internal/types"
+)
+
+type DbAuthInfo struct {
+	AuthInfo
+	user *types.UserAccount
+	org  *types.Organization
+}
+
+func (a DbAuthInfo) CurrentUser() *types.UserAccount {
+	return a.user
+}
+
+func (a DbAuthInfo) CurrentOrg() *types.Organization {
+	return a.org
+}
+
+func DbAuthenticator() authn.Authenticator[AuthInfo, AuthInfo] {
```

Comment:
I think it should be possible to change this from [AuthInfo, AuthInfo] to [AuthInfo, *DBAuthInfo] without too many changes. This would allow for removing the new functions from the AuthInfo interface and the stubs in SimpleAuthInfo that just panic